homeassistant/components/smarthab/__init__.py

```diff
@@ -26,34 +30,64 @@
 )
 
 
-def setup(hass, config) -> bool:
+async def async_setup(hass, config) -> bool:
     """Set up the SmartHab platform."""
 
+    hass.data.setdefault(DOMAIN, {})
     sh_conf = config.get(DOMAIN)
 
+    hass.async_create_task(
+        hass.config_entries.flow.async_init(
+            DOMAIN, context={"source": SOURCE_IMPORT}, data=sh_conf,
+        )
+    )
+
+    return True
+
+
+async def async_setup_entry(hass: HomeAssistantType, entry: ConfigEntry):
+    """Set up config entry for SmartHab integration."""
+
     # Assign configuration variables
-    username = sh_conf[CONF_EMAIL]
-    password = sh_conf[CONF_PASSWORD]
+    username = entry.data[CONF_EMAIL]
+    password = entry.data[CONF_PASSWORD]
 
     # Setup connection with SmartHab API
     hub = pysmarthab.SmartHab()
 
     try:
-        hub.login(username, password)
+        await hub.async_login(username, password)
     except pysmarthab.RequestFailedException as ex:
-        _LOGGER.error("Error while trying to reach SmartHab API.")
-        _LOGGER.debug(ex, exc_info=True)
-        return False
-
-    # Verify that passed in configuration works
-    if not hub.is_logged_in():
-        _LOGGER.error("Could not authenticate with SmartHab API")
+        _LOGGER.exception("Error while trying to reach SmartHab API: %s", ex)
+        raise ConfigEntryNotReady from ex
+    except Exception as ex:  # pylint: disable=broad-except
+        _LOGGER.exception("Unexpected error during login: %s", ex)
```

Comment:
```suggestion
        _LOGGER.exception("Unexpected error during login")
```
_LOGGER.exception takes care of logging the exception